lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2042,8 +2042,11 @@ static LogicalResult checkRefTypeFlow(Operation *connect) {
               "connections only";
       return diag;
     }
-    if ((src.getType().isa<RefType>() && !src.hasOneUse()) ||
-        (dst.getType().isa<RefType>() && !dst.hasOneUse())) {
+    // RefType supports multiple readers. That is, there can be multiple
+    // RefResolveOps remotely connected to a single RefSendOp.
+    // But multiple RefSendOps should not be remotely connected to a single
+    // RefResolveOp.
+    if (dst.getType().isa<RefType>() && !dst.hasOneUse()) {
```

Comment:
If this is just checking `dst`, probably can put this into the `if (dst-is-reftype)` above.

Do we still want this check?  It helps ensure `RefType`s aren't "sent" to multiple times, or connected to repeatedly, but also stops resolve from a send in same module, I think.  WDYT?

Reply:
I agree, let me move this up into the if.
You are right, that this ensures every `RefResolve` can be traced to a single `RefSend` op. But allows the multiple read case. I am not sure, both send and resolve in a single module is possible. Other constraints (downward only reference) might not allow it, unless we declare a `wire` of `RefType`. Can you come up with an example for both send/resolve in one module ?